phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixConnection.java

```diff
@@ -441,6 +453,23 @@ private static Properties filterKnownNonProperties(Properties info) {
         return result.build();
     }
 
+    public boolean isInternalConnection() {
+        return isInternalConnection;
+    }
+
+    /**
+     * This method, and *only* this method is thread safe
+     * @param connection
+     */
+    public void addChildConnection(PhoenixConnection connection) {
+        synchronized (this) {
```

Comment:
Avoid synchronizing on `this`. Let's have a separate variable which we synchronize on which only relates to the childConnections and use `@GuardedBy` syntax to denote this. Today we don't have any other methods synchronized on `this` inside PhoenixConnection, but if we do tomorrow, there will be unnecessary contention